plugins/lint/current_lint.ml

```diff
@@ -0,0 +1,86 @@
+open Current.Syntax
+module Docker = Current_docker.Default
+
+let src = Logs.Src.create "current_fmt" ~doc:"Current pipelines for linting"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let option_map f = function Some x -> Some (f x) | None -> None
+
+let version_from_string =
+  let re =
+    Re.(
+      seq
+        [
+          start;
+          rep space;
+          str "version";
+          rep space;
+          char '=';
+          rep space;
+          group (rep1 @@ diff graph (set "#"));
+          rep space;
+          eol;
+        ]
+      |> compile)
+  in
+  fun path -> Re.exec_opt re path |> option_map (fun g -> Re.Group.get g 1)
+
+let version_from_file path =
+  let ( let+ ) = Lwt.Infix.( >>= ) in
+  let ( let* ) = Lwt.Infix.( >|= ) in
```

Comment:
These are really confusing names! `let*` is supposed to be `>>=` and `let+` should be `>|=` (according to https://github.com/ocaml/ocaml/pull/1947):

> So we use let+ for map, let* for bind and set both the ands to the monoidal product.